### PR TITLE
implementation of zarr conversion

### PR DESCRIPTION
**Pull request by apatlpo**
We implement a zarr conversion capability in post that will enable:
- creating zarr archives
- deleting netcdf files

This PR lead to a major overhaul of netcdf file reading which also improves it.
By default, we do not chunk time and vertical dimensions anymore.
This was counter productive for small simulations.
For large simulations chunks will have to be provided

Hard parts: 
- one need to check for chunk sizes before storing zarr archives as it is important that these chunks are not smaller than some size (approx 4000x3000).
- add mechanisms to search for all available outputs and verify they have been stored.

The PR also proposes several the renaming of several variables (breaking changes):
- `open_nc`  -> `outputs` , note that you can pass the option 'all' now which loads all available diagnostics
- `filename` -> `nc_files`
- `suffix` -> `key`
- `chunk_time` -> `chunks` a more general parameter to control chunks when reading files.

croco vmode outputs files are excluded from the zarr conversion and deleting

### CODE REVIEW

File: crocosi/postp.py [hunk 1]
```diff
@@ -463,9 +508,79 @@ def s_coordinate(sc, theta_s, theta_b):
...
             self.grid = ds
+    
+    ### store data to zarr archives
+    def _is_zarr_archive(self, key):
+        """ Utils, test existence of a zarr archive
+        """
+        zarr_archive = join(self.zarr_dir, key+'.zarr')
+        return isdir(zarr_archive)
+
+    def store_zarr(self, chunks={}, auto_chunks=True, **kwargs):
```
Comment (NoeLahaye) [inline crocosi/postp.py:521]: what about print a few informations when doing this? Like time spent storing the archives, size of files, chunks... The process can be quite long, right?

File: crocosi/postp.py [hunk 2]
```diff
@@ -63,48 +70,46 @@ def __init__(self,
         self.dirname = os.path.expanduser(dirname)
         self.verbose = verbose
         self.prefix = prefix
-        if os.path.isfile(self.dirname+'t1/{}grid.nc'.format(prefix)):
-            _nc_default = ['grid']
-        else:
-            # for backward compatibility
-            _nc_default = ['his']
-        self.open_nc = list(set(_nc_default + open_nc))
+        self.zarr_dir = join(self.dirname, 'zarr')
```
[the remark this answers is not in the document]
Comment (apatlpo) [reply crocosi/postp.py:73]: I could do `join_path`, would that be better?

File: crocosi/postp.py (hunk 1 above)
Comment (apatlpo) [reply crocosi/postp.py:521]: time spent I'm not sure but archive size and chunk size why not, I'll think about it

File: crocosi/postp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (NoeLahaye) [reply crocosi/postp.py:73]: As you prefer. I would personally do `import os.path as path` and then use `path.join`, `path.isdir`, etc. I think there are standards but I don't know them :). I guess keeping `join` is fine

**Comment by apatlpo** [comment]
Ok, I'm going to merge this and we'll make adjustments via master.

**Pull request merged**